readme_renderer/rst.py

```diff
@@ -129,4 +129,5 @@ def render(
     if rendered:
         return clean(rendered)
     else:
+        stream.write("no output rendered")
```

Comment:
I think this could use some more work. Given a `README.rst` like:

```rst
Header
======
```

The latest (unreleased) version of Twine shows:

<img width="644" alt="Screen Shot 2022-03-13 at 9 17 31 PM" src="https://user-images.githubusercontent.com/1326704/158089029-e96ee0ef-116e-403c-aba3-1f08094b4095.png">

First, it'd be nice if the `no output rendered` message were more helpful, e.g. with a suggested resolution.

Furthermore, this new output also shows up for RST syntax errors. With a spurious header line:

```rst
Header
======

======
```

The output is:

<img width="643" alt="Screen Shot 2022-03-13 at 9 17 02 PM" src="https://user-images.githubusercontent.com/1326704/158089057-bb4061aa-14f3-4db4-a36b-24dbb2105aa3.png">

Compared to `readme_renderer==34.0`:

<img width="639" alt="Screen Shot 2022-03-13 at 9 15 58 PM" src="https://user-images.githubusercontent.com/1326704/158089071-79babad6-2aa3-46ff-8d9c-4785349a92af.png">

---

Also, I wonder if the scenario described in #149 (automatically removed header) could/should be treated the same way as an empty `README.rst`, i.e., passing with an empty description:

<img width="451" alt="Screen Shot 2022-03-13 at 9 15 47 PM" src="https://user-images.githubusercontent.com/1326704/158089216-45a27f18-3ac0-4545-a5f3-a277ee979ad8.png">

Reply:
@bhrutledge Thanks for testing this branch - these examples are very helpful!

Considering that the condition expressed in the added test case is [intentional behavior](https://github.com/pypa/readme_renderer/issues/149#issuecomment-1067261247), would it make sense to not apply this change, and rather change how twine behaves, rather than this library?
Maybe around https://github.com/pypa/twine/blob/a0ba32dcd0ea5af5de7b88d6645b7743bf003760/twine/commands/check.py#L102-L103 - check if there are warnings on the stream as well? This would then preserve the desired behavior of `readme_renderer` to emit the warnings from the docutils dependency it uses.

Then the second example with the spurious header line would still work as it's supposed to.

Reply:
Twine could check the warning stream, but if I follow this code, so could readme_renderer, and add a helpful warning if there's no content. I think I have a preference for the latter, to make this transparent to Twine, and useful in other projects that use this package. What do you think?

Reply:
Fair enough - I figured if the original report was expressing dissatisfaction with their `twine check` results, maybe twine would answer their need. 😉 

I'll give it a shot - checking the stream for *any warnings from docutils* first, and if there are, leave them be, but if there are none, then add our warning.

Do you have any suggestions for a better warning message?

Reply:
@bhrutledge Thanks for digging in - super helpful!
I've pushed another commit (after rebase of the original one on the current `main`) - which implements the clearer message you suggested, and added some smarter detection of whether or not to emit a warning.

Glad that we exposed an interesting bug on `twine` as well - but considering that with this change `readme_renderer` will emit a warning for what it considers "empty" (after removing the first Header), does the `UNKNOWN` check still make sense over there? 🤔

Reply:
> with this change `readme_renderer` will emit a warning for what it considers "empty" (after removing the first Header), does the `UNKNOWN` check still make sense over there? 🤔

I want to look into this a little more, but I think so, because otherwise readme_renderer will see `UNKNOWN\n\n`, and return `<p>UNKNOWN</p\n`. What's not clear to me at the moment is how that's rendered on PyPI, and the whole point of `twine check` is to avoid surprises in PyPI rendering.